src/utils/getDataSetsKeys.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * @flow
+ */
+
+import { List, Map, get } from 'immutable';
+import type { UUID } from 'lattice';
+
+export default function getDataSetsKeys(
+  dataSets :Map<UUID, Map>,
+  dataSetsColumns :Map<UUID, Map<UUID, Map>>,
+) :List<List<UUID>> {
+
+  return List().withMutations((keys :List<List<UUID>>) => {
+    dataSets.forEach((dataSet, dataSetId) => {
+      const dataSetColumns = dataSetsColumns.get(dataSetId);
+      keys.push(List([dataSetId]));
+      dataSetColumns.forEach((column :Map) => {
```

Comment:
Same comment. I believe `dataSetColumns` is a map from column id to column, so I think you can use `keySeq()` here as well.